removed the h3 header as not needed

DeniceLayton opened the pull request:
In this PR I have done the following:

1. Created the cat match form.
2. Created some JavaScript so that when you select options for the form  and click submit these are captured and stored in an object.
3. Created JavaScript to disable all other options when you select all cats.
4. styled the form for desktop and mobile

I would love some feedback on this good or bad as i don't have that much experience with forms and certainly styling them.  So if you have any opinions of the form, technically or maybe how it should look, or the wording then please please let me know

Commit:
removed the h3 header as not needed

Commit:
fixing conflicts with the readme file

onlyasmalllizard reviewed (changes requested):
I think the main things I would look at are the arrangement of the options and the wording. It feels a bit overwhelming to have a whole bunch of choices all on two lines. I wonder if you could visually arrange them into sections based on what need you're looking to meet. For instance, you could pair Coat Length and Active (maybe rename to something like Playfulness or Energy?) under a maintenance level. Similarly, you could put Hypoallergenic and Dog-Friendly under a "Fitting into your family" section.

I'm not clear on what low vs high means for each question. Are we rating how important it is to us (in which case, how do we specify what length of coat we want?), or how much of that thing the cat has? The header is very wordy, and I think you could move the guidance into the answer options themselves. If it's "how important this is to me?" you could have them written as "0 - Not at all important", "1 - Not very important", etc. This would let you have a much catchier header of "Search for your perfect feline friend!" (I think "Find your perfect feline friend!" might be even catchier, but that's up to you.)

I love the styling of the checkboxes when they're checked, but they're very hard to see when they're not checked. Putting a thin, dark outline would help a lot.

I feel like "Show all Cat Breeds" should be a button, not a checkbox, and near the top. My impression is that this is meant for people with no strong preferences who don't care about filling out the form. This means they shouldn't have to read through all the form options to get to the page they want to see.

If the Search button is in the centre, it needs to be a lot bigger (mostly wider) to balance out the width of the form. Alternatively, you could make it only slightly wider and put it to one side. I'd play with this after you figure out your arrangement of the options, because I think that will affect the width of the form overall.

Commit:
changed styling of the form

Commit:
changed the styling of the form

DeniceLayton commented:
Hi, 

Thanks for all your feedback on this, great suggestions its good to have another point of view! so I have done the following. 

Would you have a look over and again see what you think or what might work easier/better. I suspect this might be an evolving thing anyway once we start working with the data.

1. I have added a black outline to the checkboxes , let me know if this is easier to see or needs to be bolder/fainter. Its currently on both the checked and unchecked state. 

2. I have scrapped the idea of a checkbox for show all cats and have gone with your button idea - much better and saves the JavaScript code as well that was disabling other options if it was checked.  I'm not sure why I didn't go with a button ( talk about over complicating things!!) . So see what you think of this.

3. I liked your idea of arranging the options into sections to break them up.  have arranged it into sections within the form, let me know what you think of the spacing and wording as I couldn't quite decide. 

4. I have reworded the values so that 1= not very and 5 = very . If you look at the option i.e. dog friendly then this logic works well 1=not very dog friendly and 5 = very dog friendly.

5. I have added the "not very" and "very" options into the actual selects but I have only done it for 1 and 5 as I didn't know whether to add stuff for the middle or it would be intuitive what the in-between values were indicating. In hindsight I think I need to make up the middle two options as it would look more balanced. perhaps  " a little" and " a lot"? What do you think?

6. I have changed it so that the null value 0 does not display on the screen.

7. I have used a mobile first approach on this so styles outside of media queries have been based on galaxy fold sizes and styled accordingly , and everything in media queries targets sizes up from that.

8. I have changed the widths of the buttons, let me know if they are too big/small and need adjusting.

Let me know your thoughts

onlyasmalllizard commented:
Hi Denice, it already looks so much better with the split of categories! I have a lot of comments, but I think most of them are little things really. I'll go through your points first and anything that doesn't fit in to those I'll put at the end.

1. The outline really helps, and I don't think you have to remove it on the checked version. I think you could make them look more consistent with the dropdown menu by putting the same border outline around the menus. On the topic of visual consistency, the dropdown menus have a background colour of #f5f5f5, and the unticked checkbox has a background of #e9e9e9. I think they'd look better if you pick the same off-white for both. If you can change the border-radius for the checkbox to be similar to the dropdowns, that will help make them look more like they're meant to be on the same form.
2. I like the button! I think the padding/sizing might need adjusting for smaller laptop sizes. When I look at it on a 13" laptop, the text splits into two lines, which doesn't look great. I think if we can make sure it's on one line for at least 1200px and up, that will cover the most common laptops.
3. I think you could have a little more space between the sections. Maybe you could try to match the amount of space between the edge of the section card and the form card on the sides? I feel like maybe "Friendliness" might make more sense with "fitting into your family", since it's an interacting with others sort of characteristic. I'd also move "Intelligence" into the "Other" section, since I'm not entirely sure it affects the lifestyle, and it makes the last section not look so empty. For the last section, my two thoughts for names are either "Other" (no "options") or "Special Characteristics". I'm honestly not sure!
4. This does work, but I think you need to change it on the Coat Length one. Can you put "1 - short" and "5 - long"? Or something similar. I think you should take the scale out of the heading and have it as "Choose your options based on your needs". You explain the scale where it's relevant, so this will make it easier to read.
5. I think it is intuitive with the numbers, though I think I see it more commonly done this way on radio buttons. So in that sense maybe it will look more balanced if you write them all out. It might be fun to play with writing more descriptive selectors and dropping the numbers. As an example, "Friendliness" could have "Slow to warm up to people", "A little shy", "Confident and ready to make friends!", and "Everyone's best friend". You could go either fairly straightforward like those, or more humourous, with things like "Plotting your demise from under the bed" and "Ten different families think I'm their cat". (I am happy to help writing descriptors if you want.)
6. This was actually something I thought about last time but I don't think I said! I think this works well for two reasons. First, most non-tech people don't count from 0, so it's a little odd for a site targeting a general audience. Second, it means there is no dead middle option. It's less important in a cat-finding form, but from what I understand about survey creation, having even numbers is recommended because otherwise people will pick the middle option, which doesn't give you strong information about what they think.
7. This looks good! I wonder if it might look better to have the section card effect (the big one that's behind everything) only appear on tablets and larger. Phones have such small screens that the sides start to look a bit busy. This would also give you just a touch more space for the content.
8. The buttons seem to get very wide on the tablet settings. It also feels a little weird on some of the screen sizes that the search button is almost as wide as the card above it but not quite, even though I think that is too wide for a button in this context! For the Search button, I think it will look normal for a form to have it smaller with its right side aligned with the right edge of the card above it. For the "Show all cats" button, on large screens, you could actually move the text "I don't want to search, show me all cat breeds", which would make the width of button not look so wide. Then on smaller screens, you could have it as it is now. I think this button should still be centred. It's serving a different purpose on the page than the final button, so I don't think you need to line them up with each other. I have a little more to say on the sizing, but it will fit better into one of my coming notes.

Here are my additional thoughts that aren't a direct response to your notes.

I think the scale is a little off between elements. The "Choose your options" header feels like it should be bigger than the card headers. The tick boxes look tiny on tablet sizes and above. The buttons are as big or bigger than the main header on some screen sizes. I think if you go through and work out one to one comparisons between elements (i.e. "Element 1 should be larger than/smaller than/the same size as element 2"), it will give you a clear guideline of how to size everything on the page.

There's a lot of white space between the labels and the dropdowns/tickboxes on larger screens. With the dropdown menus, it might look nice to make them wider and have them left-aligned underneath their respective labels, instead of to the side. This would also give you more space if you want to write more descriptive options. I'm not entirely sure how to place the tick boxes without having a play. It might look more obvious once you sort out the dropdown menus. Or maybe there is a better solution than tickboxes. One thought is making the words for the tickbox options toggle-able. Or you could have a yes/no dropdown? I'm sure you'll find a good solution if you have a play :)

I would remove the card design from behind the "Show all cats" section. It's not a section of the form the same way the other ones are, and it looks a little cramped on some screen sizes.

I'd remove the full stops from the main header and sentence underneath. I think you could replace both or either with exclamation marks. If you replace the "show all cats" with an exclamation mark, I'd have it punctuated as "I don't want to search. Show me all the cat breeds!"

Overall, looking at this has made me really excited for how the page is going to turn out. I can really see the style direction you're going in!

Commit:
fixing merge conflicts
